### PR TITLE
refactor test_error_injection.py into a unittest.TestCase class

### PR DESCRIPTION
**Pull request by thrau**
this PR refactors the test methods in https://github.com/localstack/localstack/blob/master/tests/integration/test_error_injection.py  into a unittest.TestCase class.

### CODE REVIEW

**Comment by GonzaloSaad** [comment]
Question! Do you favor class-styled tests (`unittest` suites) againts functional-styled tests (`pytest` suites)?

**Comment by thrau** [comment]
> Question! Do you favor class-styled tests (`unittest` suites) againts functional-styled tests (`pytest` suites)?

since almost all of the tests use class-style unittest tests, and they're compatible with pytest, i think it makes sense to use them.

do you have an argument for the functional-styled tests? are some features of pytest like fixtures better supported?

**Comment by GonzaloSaad** [comment]
> > Question! Do you favor class-styled tests (`unittest` suites) againts functional-styled tests (`pytest` suites)?
> 
> since almost all of the tests use class-style unittest tests, and they're compatible with pytest, i think it makes sense to use them.

Cool! Understood! Asking because in some moment I'll have to write them and wanted to know! 

> do you have an argument for the functional-styled tests? are some features of pytest like fixtures better supported?

Not an objective one! `pytest` supports `unittest` so the most used features apply the same. 

Thanks for the answer :raised_hands: !

File: tests/integration/test_error_injection.py [hunk 1]
```diff
@@ -1,117 +1,109 @@
...
+        partition_key = short_uid()
+        self.assert_zero_probability_read_error_injection(table, partition_key)
 
-    # with a probability of 1, always throw errors
-    config.DYNAMODB_READ_ERROR_PROBABILITY = 1.0
-    with pytest.raises(ClientError):
-        table.get_item(Key={PARTITION_KEY: partition_key})
+        # with a probability of 1, always throw errors
+        config.DYNAMODB_ERROR_PROBABILITY = 1.0
+        with pytest.raises(ClientError):
```
Comment (whummer) [inline tests/integration/test_error_injection.py:71]: nit: we could also use the `unittest` variant of this (not critical, shouldn't hold back the merge):
```
with self.assertRaises(ClientError):
    ...
```

**Review by whummer — APPROVED** [review]
Great, thanks @thrau ! 🎉

File: tests/integration/test_error_injection.py (hunk 1 above)
Comment (thrau) [reply tests/integration/test_error_injection.py:71]: good point! much cleaner

**Pull request merged**